File: src/main/java/com/_200studios/learn/api/allowed_users/AllowedUserService.java
package com._200studios.learn.api.allowed_users;

import com._200studios.learn.api.allowed_users.dto.AllowedUserCreateDto;
import com._200studios.learn.shared.types.GlobalRole;
import org.springframework.stereotype.Component;

@Component
public class AllowedUserService {
    private final AllowedUserRepository allowedUserRepository;

    public AllowedUserService(AllowedUserRepository allowedUserRepository) {
        this.allowedUserRepository = allowedUserRepository;
    }

    public AllowedUser create(AllowedUserCreateDto dto) {
        AllowedUser user = new AllowedUser();
        user.setEmail(dto.getEmail());
        user.setGlobalRole(GlobalRole.STUDENT);

        return allowedUserRepository.save(user);
    }
}

File: src/api/allowed-users/allowed-user.service.ts
import { AllowedUserCreateDto } from '@/api/allowed-users/dto/allowed-user-create.dto';
import { AllowedUser } from '@/api/allowed-users/models/allowed-user.model';
import { PrismaService } from '@/shared/datasource/prisma/prisma.service';
import { Injectable } from '@nestjs/common';
import { GlobalRole } from '@prisma/local-client';

@Injectable()
export class AllowedUserService {
  constructor(private readonly prismaService: PrismaService) {}

  public async findByEmail(email: string): Promise<AllowedUser> {
    return this.prismaService.allowedUser.findUnique({
      where: {
        email,
      },
    });
  }

  public async create(data: AllowedUserCreateDto): Promise<AllowedUser> {
    return this.prismaService.allowedUser.create({
      data: {
        ...data,
        globalRole: GlobalRole.STUDENT,
      },
    });
  }
}
